driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -175,18 +176,16 @@ public void resetTimeout() {
     }
 
     public TimeoutContext withAdditionalReadTimeout(final int additionalReadTimeout) {
+        // Only used outside timeoutMS usage
+        assertNull(timeout);
+
         // Check existing read timeout is infinite
-        if (getReadTimeoutMS() == 0) {
+        if (timeoutSettings.getReadTimeoutMS() == 0) {
             return this;
         }
 
         long newReadTimeout = getReadTimeoutMS() + additionalReadTimeout;
-        newReadTimeout = newReadTimeout > 0 ? newReadTimeout : Long.MAX_VALUE;
-        if (timeout != null) {
-            return new TimeoutContext(timeoutSettings.withTimeoutMS(newReadTimeout));
-        } else {
-            return new TimeoutContext(timeoutSettings.withReadTimeoutMS(newReadTimeout));
-        }
+        return new TimeoutContext(timeoutSettings.withReadTimeoutMS(newReadTimeout > 0 ? newReadTimeout : Long.MAX_VALUE));
```

Comment:
[optional] In the code above, we checked that `readTimeout` is not 0, implying it could be positive or negative. In the ternary operation, is there an assumption that `getReadTimeoutMS()` or `additionalReadTimeout `might be negative? If negative values are not expected, i think we should add an assertion at the beginning to catch any misconfiguration.

Reply:
Done